lib/talkable/generators/install_generator.rb

```diff
@@ -0,0 +1,56 @@
+module Talkable
+  class InstallGenerator < Rails::Generators::Base
+    source_root File.expand_path("../templates", __FILE__)
+    class_option :haml, type: :boolean, default: false
+    class_option :slim, type: :boolean, default: false
+
+    def ask_config_values
+      @site_slug  = ask("Your Talkable site slug:")
+      @api_key    = ask("Your Talkable API Key:")
+      if ['', 'y', 'Y'].include?(ask 'Do you have a custom domain? [Y/n]')
+        @server   = ask("Your custom domain [#{Talkable::Configuration::DEFAULT_SERVER}]:")
+      end
+    end
+
+    def add_initializer
+      template "config/initializers/talkable.rb", "config/initializers/talkable.rb"
+    end
+
+    def inject_talkable_offer
+      inject_into_file "app/controllers/application_controller.rb", after: "protect_from_forgery with: :exception" do
+<<-RUBY
+
+  before_action :load_talkable_offer
+RUBY
+      end
+
+      inject_into_file "app/controllers/application_controller.rb", before: /^end/ do
+<<-RUBY
+  protected
+
+  def load_talkable_offer
+    origin = Talkable.register_affiliate_member(campaign_tags: 'popup')
+    @offer ||= origin.offer if origin
+  end
+RUBY
+      end
+
+      empty_directory "app/views/shared/"
+
+      if options[:haml] || options[:slim]
```

Comment:
I think you can guess which template engine to use by looking at the extension of file like `app/views/layouts/application.html.xxx`